grammars/css.cson

```diff
@@ -20,294 +20,1205 @@
       {
-        'include': '#string-single'
+        'begin': '(?i)(?<![\\w-])(rgba?|hsla?)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'support.function.misc.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.color.${1:/downcase}-value.css'
+        'patterns': [
+          {
+            'include': '#property-values'
+          }
+        ]
       }
+      # Gradients
       {
-        'include': '#comment-block'
+        'begin': '''(?xi) (?<![\\w-])
+          (
+            (                            # Captured for scope-name
+              (?:-webkit-|-moz-|-o-)?    # Accept prefixed/historical variants
+              (?:repeating-)?            # "Repeating"-type gradient
+              (?:linear|radial|conic)    # Shape
+            )
+            -gradient
+          )
+          (\\()
+        '''
+        'beginCaptures':
+          '1':
+            'name': 'support.function.gradient.css'
+          '3':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.${2:/downcase}.gradient.css'
+        'patterns': [
+          {
+            'match': '(?i)(?<![\\w-])(from|to|at)(?![\\w-])'
+            'name': 'keyword.operator.gradient.css'
+          }
+          {
+            'include': '#property-values'
+          }
+        ]
+      }
+      # Gradients (Deprecated/original Webkit syntax)
+      {
+        'begin': '(?i)(?<![\\w-])(-webkit-gradient)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'invalid.deprecated.gradient.function.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.legacy-webkit.gradient.css'
+        'patterns': [
+          {
+            'begin': '(?i)(?<![\\w-])(from|to|color-stop)(\\()'
+            'beginCaptures':
+              '1':
+                'name': 'invalid.deprecated.function.css'
+              '2':
+                'name': 'punctuation.section.function.begin.bracket.round.css'
+            'end': '\\)'
+            'endCaptures':
+              '0':
+                'name': 'punctuation.section.function.end.bracket.round.css'
+            'patterns': [
+              {
+                'include': '#property-values'
+              }
+            ]
+          }
+          {
+            'include': '#property-values'
+          }
+        ]
+      }
+      # Misc functions
+      {
+        'begin': '''(?xi) (?<![\\w-])
+          ( annotation
```

Comment:
This can definitely be condensed.  Readability is good, but it doesn't have to be just one per line.